pkg/version/discover.go

```diff
@@ -25,7 +25,27 @@ import (
 	"github.com/k8stopologyawareschedwg/deployer/pkg/deployer/platform/detect"
 )
 
-func DiscoverCluster(ctx context.Context, platformName, platformVersion string) (platform.Platform, platform.Version, error) {
+// DiscoverResult is the result of the cluster discovery process of platform and version.
+// It stores the platform, the short version (minimized) and the long version (full).
+// the short version is the version without the build metadata, and is the one for
+// general use. For more specific use cases concerning to specific Kubernetes versions,
+// the long version should be used.
+type DiscoverResult struct {
+	Platform     platform.Platform
+	ShortVersion platform.Version
+	LongVersion  platform.Version
+}
+
+func newUndiscoveredClusterResult() DiscoverResult {
```

Comment:
The name is a bit clunky, **Undiscovered**ClusterResult?
`NewClusterResult` which returns ClusterResult with initialized but undefined/default values is a common behavior.

Reply:
Or even `NewDefaultClusterResult`